### ISSUE TITLE
Using an invalid source URI should first check for validity before resetting existing selection/mapping

### ISSUE THREAD

**Issue by bsrikan** [marklogic/marklogic-data-hub#1181]
#### The issue
Using an invalid source URI resets the existing selection/mapping

Short description of the problem:
1. Say you are changing the source URI of an already saved map.
2. Use an invalid Source URI and click on OK in the dialog box that prompts it will remove existing property selections

Actual Behavior:
After step 2 existing selection is reset and only then source URI is validated

What behavior are you expecting?
Source URI should be validated first before resetting/removing existing property selections.
The workflow should be updated to be:
```
Update source URI by selecting green tick button -> Check for URI validity ->
           isValid -> present msg box with OK/CANCEL saying it will remove existing property selections
                      onOK -> remove selections, update URI
                      onCancel -> bring back to last saved state
           isNotValid -> present msg box with OK saying Doc URI not found
                      onOK -> bring back to last saved state(same as onCancel)
```
#### Tech details

Which Operating System are you using?
Mac
Which version of MarkLogic are you using?
90 nightly
Which version of the Data Hub Framework are you using?
3.1.0
Is this a QuickStart UI bug? If so, which browser are you using?
Yes but not browser specific. Its an implementation bug.

┆Status: In QA

**Comment by wooldridge** [comment]
Thanks, @bsrikan. 

This is a duplicate of: https://project.marklogic.com/jira/browse/DHFPROD-1025

I will close the above issue and leave this one, yours.
